sunpy/io/meta/fits_meta.py

```diff
@@ -0,0 +1,655 @@
+        missing_keywords = set(required_keywords).difference(set(these_keywords))
+        if len(missing_keywords) > 0:
+            warn_user(f"These required keywords {missing_keywords} are missing.")
+
+        for this_card in self.cards:
+            if this_card.value == '':  # check that all keywords have values.
+                warn_user(f'{this_card} has no value.')
+            if this_card.keyword.upper() in schema['keyword']:
+                schema_params = schema.loc[this_card.keyword]
+                # check that the required description is present
+                if schema_params['description'] != this_card.comment:
+                    warn_user(f'{this_card.comment} does not match {schema_params["description"]}')
+
+                match schema_params['data_type']:
+                    case "quantity":
+                        #  get the unit from the comment which should be between []
+                        # TODO: update this expression to remove the brackets
+                        unit_list = re.findall(r'\[.*?\]', this_card.comment)
+                        if len(unit_list) == 0:
+                            warn_user(f"No unit found in {this_card.comment}")
+                        if len(unit_list) > 1:
+                            warn_user(f"Too many units found in {this_card}")
+                        if len(unit_list) == 1:
+                            # test that the unit is readable
+                            # TODO: if brackets are remove above then fix the line below
+                            unit_str = unit_list[0][1:-1]  # strip the brackets
+                            try:
+                                u.Unit(unit_str)
+                            except ValueError:
+                                warn_user(f"{unit_str} is not a recognized unit.")
+                    case 'date':
+                        try:
+                            parse_time(this_card.value)
+                        except ValueError:
+                            warn_user(f"Date in {this_card.value} is not recognized.")
+                    case 'float':
+                        try:
+                            float(this_card.value)
+                        except ValueError:
+                            warn_user(f'The value in {this_card} cannot be interpreted as a float')
+
+    def add_observer_keywords(observer_coordinate):
+        pass
+
+    def add_wcs_keywords():
+        pass
+
+    def _check_unit(card):
+        """Given a card, check that its value is recognized by astropy units 
+        or generate a warning."""
+        try:
+            u.Unit(card.value)
+        except ValueError:
+            warn_user(f'Value in {card.keyword} is not a recognized unit.')
+
+    def check_for_deprecated_keywords(self):
+        """Check if any keywords are deprecated."""
+        result = False
+        deprecated_keywords = ['DATE-OBS', 'EXPTIME']
+        replacement_keywords = ['DATE-BEG', 'XPOSURE']
```

Comment:
These should probably live as a property on the class and be a dictionary (key is the deprecated keyword, value is the replacement keyword).